src/promptflow-evals/promptflow/evals/synthetic/adversarial_simulator.py

```diff
@@ -0,0 +1,316 @@
+)
+from ._utils import JsonLineList
+
+logger = logging.getLogger(__name__)
+
+
+class AdversarialSimulator:
+    @monitor_operation(activity_name="adversarial.simulator.init", activity_type=ActivityType.PUBLICAPI)
+    def __init__(self, *, azure_ai_project: Dict[str, Any]):
+        """
+        Initializes the adversarial simulator with a project scope.
+
+        :param azure_ai_project: Dictionary defining the scope of the project. It must include the following keys:
+            - "subscription_id": Azure subscription ID.
+            - "resource_group_name": Name of the Azure resource group.
+            - "project_name": Name of the Azure Machine Learning workspace.
+            - "credential": Azure credentials object for authentication.
+        :type azure_ai_project: Dict[str, Any]
+        """
+        # check if azure_ai_project has the keys: subscription_id, resource_group_name, project_name, credential
+        if not all(
+            key in azure_ai_project for key in ["subscription_id", "resource_group_name", "project_name", "credential"]
+        ):
+            raise ValueError(
+                "azure_ai_project must contain keys: subscription_id, resource_group_name, project_name, credential"
+            )
+        # check the value of the keys in azure_ai_project is not none
+        if not all(
+            azure_ai_project[key] for key in ["subscription_id", "resource_group_name", "project_name", "credential"]
+        ):
+            raise ValueError("subscription_id, resource_group_name, project_name, and credential must not be None")
+        self.azure_ai_project = azure_ai_project
+        self.token_manager = ManagedIdentityAPITokenManager(
+            token_scope=TokenScope.DEFAULT_AZURE_MANAGEMENT,
+            logger=logging.getLogger("AdversarialSimulator"),
+        )
+        self.rai_client = RAIClient(azure_ai_project=azure_ai_project, token_manager=self.token_manager)
+        self.adversarial_template_handler = AdversarialTemplateHandler(
+            azure_ai_project=azure_ai_project, rai_client=self.rai_client
+        )
+
+    def _ensure_service_dependencies(self):
+        if self.rai_client is None:
+            raise ValueError("Simulation options require rai services but ai client is not provided.")
+
+    @monitor_operation(activity_name="adversarial.simulator.call", activity_type=ActivityType.PUBLICAPI)
+    async def __call__(
+        self,
+        *,
+        scenario: str,
+        target: Callable,
+        max_conversation_turns: int = 1,
+        max_simulation_results: int = 3,
+        api_call_retry_limit: int = 3,
+        api_call_retry_sleep_sec: int = 1,
+        api_call_delay_sec: int = 0,
+        concurrent_async_task: int = 3,
+        jailbreak: bool = False,
+    ):
+        """
+        Executes the adversarial simulation against a specified target function asynchronously.
+
+        :param scenario: Scenario string used for generating adversarial inputs.

```

Comment:
doc or doc link on the available scenarios?

Reply:
@minthigpen can you share the link once the docs are published?

Reply:
@wangchao1230  - all the scenarios will be in the docs, will share link once it is published.